### ISSUE TITLE
pgbackrest s3 doesn't work 

### ISSUE THREAD

**Issue by lkhomenk** [CrunchyData/postgres-operator#987]
**Describe the bug**
S3 bckup doesn't work with `--pgbackrest=true --pgbackrest-storage-type=s3` .
Primary repo pgbackrest config 
```
PGBACKREST_REPO1_HOST=test-rest-backrest-shared-repo
PGBACKREST_REPO1_S3_BUCKET=<bucket_name>
PGBACKREST_STANZA=db
PGBACKREST=true
PGBACKREST_REPO_TYPE=s3
PGBACKREST_REPO1_S3_KEY_SECRET=<deducted>
PGBACKREST_REPO1_S3_REGION=<region>
PGBACKREST_DB_PATH=/pgdata/test-rest
PGBACKREST_REPO1_S3_CA_FILE=/sshd/aws-s3-ca.crt
PGBACKREST_LOG_PATH=/tmp
PGBACKREST_REPO1_S3_ENDPOINT=<endpoint>
PGBACKREST_PG1_PORT=5432
PGBACKREST_REPO1_PATH=/backrestrepo/test-rest-backrest-shared-repo
PGBACKREST_REPO1_S3_KEY=<deducted>
```
leads to pgbackrest [039] error code, while primary tries to push archives to the `PGBACKREST_REPO1_HOST=test-rest-backrest-shared-repo`. removing repo1-host makes pgbackrest work
**To Reproduce**
1. provide aws key with access to your bucket(fill postgres-operator/conf/pgo-backrest-repo/aws-s3-credentials.yaml )
2. create cluster with `--pgbackrest=true --pgbackrest-storage-type=s3`
3. check primary pod logs
4. example of failure
```
pgbackrest info
ERROR: [039]: raised from remote-0 protocol on 'test-rest-backrest-shared-repo': S3 request failed with 403: Forbidden
       *** URI/Query ***:
       /?delimiter=0.000000&list-type=2&prefix=backrestrepo0.000000test-rest-backrest-shared-repo0.000000backup0.000000
       *** Request Headers ***:
       authorization: <redacted>
       content-length: 0
       host: <bucket>.<endpoint>.amazonaws.com
       x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
       x-amz-date: <redacted>
       *** Response Headers ***:
       content-type: application/xml
       date: Tue, 10 Sep 2019 14:31:25 GMT
       server: AmazonS3
       transfer-encoding: chunked
       x-amz-bucket-region: <region>
       x-amz-id-2: re0bo3IhbN14vnGwTqPJjEA/Sh6tTsbCFaHxmva5ubfC2SI2Tr/RV1CxZ67FxFZVCx57u3OR/Tk=
       x-amz-request-id: F784FD078BE74523
       *** Response Content ***:
       <?xml version="1.0" encoding="UTF-8"?>
       <Error><Code>InvalidAccessKeyId</Code><Message>The AWS Access Key Id you provided does not exist in our records.</Message><AWSAccessKeyId>(null)</AWSAccessKeyId><RequestId>F784FD078BE74523</RequestId><HostId>re0bo3IhbN14vnGwTqPJjEA/Sh6tTsbCFaHxmva5ubfC2SI2Tr/RV1CxZ67FxFZVCx57u3OR/Tk=</HostId></Error>
```
5.  unset repo1-host
```
unset PGBACKREST_REPO1_HOST
sh-4.2$ pgbackrest info
No stanzas exist in the repository.
```
**Expected behavior**
Working backup

**Please tell us about your environment:**

* Container Image Tag: centos7-10.10-2.4.1
* Platform Version: 4.0.1

**Additional context**
Guide tells that both S3 and local can be enabled simultaneously, but it seems local is enabled when s3 only is requested

**Comment by lkhomenk** [comment]
pgbackrest commands can be executed successfully from the shared-repo-pod. which doesn't have repo1-host. it seems like primary tries to push archives into shared-repo pod(local backups) and shared-backrest-repo pod pushes them to the S3. But why? only S3 was requested

**Comment by lkhomenk** [comment]
https://github.com/CrunchyData/postgres-operator/blob/7fb5f61381314e0f7886709a6ffb27c027571050/operator/cluster/clusterlogic.go#L97 - creates backrest-shared-repo deployment anyway, no matter what was requested, this deployment doesn't get `PGBACKREST_REPO_TYPE` while it gets only `conf/postgres-operator/pgbackrest-s3-env-vars.json` varibles. which makes backrest-shared-repo work as local backup storage.

**Comment by andrewlecuyer** [comment]
@lkhomenk as you are seeing, with the current Operator design a dedicated repository is created and utilized regardless - even if S3 only is selected for pgBackRest storage.  And when a dedicated repository is configured for the primary DB (i.e. `PGBACKREST_REPO1_HOST` is specified in the primary pod) along with S3 for storage, pgBackRest pushes archives to S3 using the shared repo (as you saw).  This allows only the dedicated pgBackRest repository to be exposed externally to S3.

As for the error you are seeing, this appears to be a known issue that has been addressed in #872.

**Comment by jkatz** [comment]
I would be remiss to add that pgBackRest support, let alone S3 support, received a major improvement in v4.2.0 as we standardized the PostgreSQL Operator physical backups around pgBackRest: https://github.com/CrunchyData/postgres-operator/releases/tag/v4.2.0
